CONTRIBUTING.md

```diff
@@ -44,170 +64,249 @@ not to be able to mint negative values
+   ```
 
-Set the `LOG_FILE_PATH` environment variable to an appropriate location to store the logs and parse them using the above packages.
+4. Start the description using imperative verb in present tense. Examples: Implement, Add, Update.
 
-## Style Guides
+</details>
 
 ### Git Style Guide
 
--  We require you to **Sign-off every commit**. If you don't [DCO](https://github.com/apps/dco) will not let you merge. Please add: `Signed-off-by: $NAME <$EMAIL>` as the final line of your commit message. You can do it automatically using `git commit -s`
--  It's also useful to format your commit messages appropriately. Try to keep them short.
--  Use the PR categories as a template for your commit messages: e.g. `[fix] #1969: Fix regression in Kotlin SDK tests`.
--  **Prefer present tense** ("Add feature", not "Added feature").
--  **Prefer imperative mood** ("Deploy to docker..." not "Deploys to docker...").
--  Write a meaningful commit message. Imagine that you're looking at someone's code, titled "Fixed error".
--  Limit the first line of your commit message to 50 characters or less.
--  The first line of your commit message should contain the summary of the work you've done. If you need more than one line, leave a blank line between each paragraph and describe your changes in the middle. The last line must be the sign-off.
--  Use the [Git Rebase Workflow](https://git-rebase.io/). Avoid using `git pull` use `git pull --rebase` instead.
--  If you modify the Schema (check by generating the schema with `kagami schema` and diff), you should make all changes to the schema in a separate commit with the message `[schema]`.
--  Generally, try to stick to one commit per meaningful change.
-  -  If you fixed several issues in one PR, give them separate commits.
-  -  As mentioned previously changes to the `schema` and the API should be done in appropriate commits separate from the rest of your work.
-  -  Don't bother with separate commits for fixing review comments. Amend the last one, unless the review comment asks to change the `schema`-affecting work. In that case, you want to rebase interactively.
-  -  Tests for functionality in the same commit as the functionality.
+- [Fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) the [repository](https://github.com/hyperledger/iroha/tree/iroha2-dev) and [create a feature branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository) for your contributions.
+- Use the [Git Rebase Workflow](https://git-rebase.io/). Avoid using `git pull`. Use `git pull --rebase` instead.
+
+Follow these commit guidelines:
+
+- **Sign-off every commit**. If you don't, [DCO](https://github.com/apps/dco) will not let you merge.
+  
+  You may use `git commit -s` to automatically add `Signed-off-by: $NAME <$EMAIL>` as the final line of your commit message.
```

Comment:
Might make sense to also add that both the name and e-mail should be exactly the same as listed in the contributor's github account, otherwise later we run into some new problems with DCO when we merge from `iroha2-dev` into `iroha2` as we recently learned. Also could mention that signing with GPG is welcome (`git commit -S`).

Reply:
```suggestion
  You should use `git commit -s` to automatically add `Signed-off-by: $NAME <$EMAIL>` as the final line of your commit message.
  
  Also note that when you use `git config --global` often first-time users forget to quote their entire name, and due to globbing in shells like `bash`, `zsh` and `fish`, only the first name is actually added to the sign-off. 
  
  Also, we would appreciate if your name in the sign-off matched the username on GitHub. Having a mismatch of the type "Sam H. Smith" vs. "Sam Smith" is OK, (although annoying), while a mismatch "L33t H4xxor" vs. "John Smith" is problematic.
```

Reply:
updated